tests/functionnal/KnowbaseItem.php

```diff
@@ -345,39 +345,228 @@ public function testGetForCategory()
+                    //Find rows that contain the words “apple” and "juice", or “apple” and "macintosh" (in any order), but rank “apple macintosh" higher than “apple juice".
+                ],
+                'type' => 'search',
+                'count' => 2,
+                'sort' => ['_knowbaseitem02', '_knowbaseitem01'],
+            ],
+            [
+                'params' => [
+                    'knowbaseitemcategories_id' => 0,
+                    'faq' => false,
+                    'contains' => "Know*",
+                    //Find rows that contain "Know" such as "Knowledge"
+                ],
+                'type' => 'search',
+                'count' => 2,
+                'sort' => ['_knowbaseitem01', '_knowbaseitem02'],
+            ],
+            [
+                'params' => [
+                    'knowbaseitemcategories_id' => 0,
+                    'faq' => false,
+                    'contains' => "turn*",
+                    //Find rows that contain "turn" such as "turnover"
+                ],
+                'type' => 'search',
+                'count' => 1,
+                'sort' => ['_knowbaseitem01'],
+            ],
+            [
+                'params' => [
+                    'knowbaseitemcategories_id' => 0,
+                    'faq' => false,
+                    'contains' => '"macintosh strudel"',
+                    //Find rows that contain the exact phrase “macintosh strudel”
+                ],
+                'type' => 'search',
+                'count' => 1,
+                'sort' => ['_knowbaseitem02'],
+            ],
+            [
+                'params' => [
+                    'knowbaseitemcategories_id' => 0,
+                    'faq' => false,
+                    'contains' => '"base entry _knowbaseitem02"',
+                    //Find rows that contain the exact phrase “base entry _knowbaseitem02”
+                ],
+                'type' => 'search',
+                'count' => 1,
+                'sort' => ['_knowbaseitem02'],
             ]
         ];
     }
 
     /**
      * @dataprovider testGetListRequestProvider
      */
-    public function testGetListRequest(array $params, string $type): void
+    public function testGetListRequest(array $params, string $type, int $count, mixed $sort): void
     {
         global $DB;
+        $this->login(); //to prevent KnowBaseItem entity restrict criteria for anonymous user
 
-       // Build criteria array
+        // Build criteria array
         $criteria = \KnowbaseItem::getListRequest($params, $type);
         $this->array($criteria);
 
-       // Check that the request is valid
-        $DB->request($criteria);
+        // Check that the request is valid
+        $iterator = $DB->request($criteria);
+
+        //count KnowBaseItem found
+        $this->integer($iterator->numrows())->isEqualTo($count);
+
+        // check order if needed
+        if ($sort != null) {
```

Comment:
Maybe something like that would be simpler and scale with any sized array.

```php
$names = array_column(iterator_to_array($iterator), 'name');
$this->array($names)->isEqualTo($sort);
```

Otherwise very good 👍

Reply:
done =)